paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkConnectorOptions.java

```diff
@@ -224,6 +224,18 @@ public class FlinkConnectorOptions {
                     .withDescription(
                             "If the new snapshot has not been generated when the checkpoint starts to trigger, the enumerator will block the checkpoint and wait for the new snapshot. Set the maximum waiting time to avoid infinite waiting, if timeout, the checkpoint will fail. Note that it should be set smaller than the checkpoint timeout.");
 
+    public static final ConfigOption<Boolean> LOOKUP_ASYNC =
+            ConfigOptions.key("lookup.async")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription("Whether to enable async lookup join.");
+
+    public static final ConfigOption<Integer> LOOKUP_ASYNC_THREAD_NUMBER =
+            ConfigOptions.key("lookup.async-thread-number")
+                    .intType()
+                    .defaultValue(16)
```

Comment:
I remember that you ever mentioned that the number of async threads cannot be larger than 1 as the rocksDB cache is not thread-safe.

Reply:
So there is a lock to lookup.

This parameter is due to the implementation of Flink. Currently, in the case of join missing, the Flink implementation will sleep and wait, which will cause the thread to be blocked. So we need more threads to ensure allow_unordered availability.

See `RetryableAsyncLookupFunctionDelegator.lookupWithRetry` in Flink.